add extra_template back

flyinprogrammer opened the pull request:
Fix regression in https://github.com/DataDog/puppet-datadog-agent/pull/261
where an `extra_template` was no longer getting concat'd to the template file.

truthbk commented:
@flyinprogrammer first of all, apologies for the regression and thank you for this awesome fix. This is ready to go, but I'd like to ask you for a favor.... Could you add a spec test for this? Just to ensure both the extra template and the apm footer are dropped in correctly (with and without an extra template being defined).

Let me know if you don't have the bandwidth

flyinprogrammer commented:
yeah let's see what i can get done with this tonight

flyinprogrammer commented on Gemfile:
```diff
@@ -18,4 +18,5 @@ group :development do
   gem "puppet-blacksmith"
   gem "nokogiri", "~> 1.6.0"
   gem "guard-rake"
+  gem "xmlrpc"
```
Otherwise I kept getting:

```
Could not autoload puppet/type/package: Could not autoload puppet/provider/package/pip: cannot load such file -- xmlrpc/client
```

when running the tests.

flyinprogrammer replied:
```
xmlrpc-0.3.0 requires ruby version >= 2.3, which is incompatible with the
current version, ruby 2.0.0p648
```

this is why i write golang now. hld plz.

flyinprogrammer commented on spec/classes/datadog_agent_spec.rb:
```diff
@@ -593,6 +596,20 @@
             end
+            context 'with extra_template enabled' do
+                let(:params) {
+                    {:extra_template => 'custom_datadog/extra_template_test.erb',
```
I think I added this fixture correctly, I know ignore this directory in `.gitignore`, but it seemed like the correct thing to do? Please advise!

truthbk replied:
This might be the right thing to do.... let me think about it for a sec :)

flyinprogrammer commented:
idk how you want to fix:

```
Requested binary installation but no rubies are available to download, consider skipping --binary flag.
Gemset '' does not exist, 'rvm ruby-2.1.0 do rvm gemset create ' first, or append '--create'.
```

truthbk commented:
@flyinprogrammer looks like a problem with the current test matrix on Travis. Looking into it. Thanks a bunch for the tests 💯

truthbk commented:
Fixed the Travis issue, could you please rebase to the latest `master`?

truthbk commented on Gemfile:
```diff
@@ -18,4 +18,5 @@ group :development do
   gem "puppet-blacksmith"
   gem "nokogiri", "~> 1.6.0"
   gem "guard-rake"
+  gem "xmlrpc" if RUBY_VERSION >= '2.3'
```
Could you elaborate a bit on this (for posterity)?

flyinprogrammer replied:
locally I'm running the tests against ruby `2.4.1p111` and in order to get the tests to pass you need to install this gem. of course when your ruby is < `2.3` the builds will barf apparently.

truthbk commented:
OK - so this is how I addressed the issue:
```
diff --git a/.fixtures.yml b/.fixtures.yml
index 21cff47..5d7a8ec 100644
--- a/.fixtures.yml
+++ b/.fixtures.yml
@@ -9,4 +9,5 @@ fixtures:
     ruby: "git://github.com/puppetlabs/puppetlabs-ruby.git"
     remote_file: "git://github.com/lwf/puppet-remote_file.git"
   symlinks:
+    custom_datadog: "#{source_dir}/spec/custom_fixtures/custom_datadog"
     datadog_agent: "#{source_dir}"
diff --git a/Gemfile b/Gemfile
index 5e72eb7..11ccd1a 100644
--- a/Gemfile
+++ b/Gemfile
@@ -3,6 +3,7 @@ source "https://rubygems.org"
 group :test do
   gem "syck"
   gem "safe_yaml", "~> 1.0.4"
+  gem "hiera-eyaml"
   gem "listen", "~> 3.0.0"
   gem "puppet", ENV['PUPPET_VERSION'] || '~> 4.2.0'
   gem "puppet-lint"
diff --git a/spec/fixtures/modules/custom_datadog/templates/extra_template_test.erb b/spec/custom_fixtures/custom_datadog/templates/extra_template_test.erb
similarity index 100%
rename from spec/fixtures/modules/custom_datadog/templates/extra_template_test.erb
rename to spec/custom_fixtures/custom_datadog/templates/extra_template_test.erb
```

I copied `spec/fixtures/modules/custom_datadog/templates/extra_template_test.erb` to `spec/custom_fixtures/custom_datadog/templates/extra_template_test.erb` since `spec/fixtures` is considered ephemeral.

And I then fixed up `.fixtures.yml` as you can see in the diff. I had to add the `hiera-eyaml` change to gemfil, not sure if you might need it too to get the tests to fly.

Let me know if you don't like this approach. :)

flyinprogrammer commented:
I think this is great and I think you're great @truthbk !! Thanks for teaching me how to do the fixtures, and I'm sorry for kinda being lazy and not digging into the docs more. I rebased off master and squashed my commits.
